libcloud/compute/drivers/gig_g8.py

```diff
@@ -0,0 +1,501 @@
+        networkid = self._api_request("/cloudspaces/create", params)
+        network = self._api_request("/cloudspaces/get",
+                                    {"cloudspaceId": networkid})
+        return self._to_network(network)
+
+    def ex_destroy_network(self, network):
+        self._api_request("/cloudspaces/delete",
+                          {"cloudspaceId": int(network.id)})
+        return True
+
+    def stop_node(self, node):
+        """
+        Stop virtual machine
+        """
+        node.state = NodeState.STOPPING
+        self._api_request("/machines/stop", {"machineId": int(node.id)})
+        node.state = NodeState.STOPPED
+        return True
+
+    def ex_list_portforwards(self, network):
+        data = self._api_request("/portforwarding/list",
+                                 {"cloudspaceId": int(network.id)})
+        forwards = []
+        for forward in data:
+            forwards.append(self._to_port_forward(forward, network))
+        return forwards
+
+    def ex_create_portforward(self, network, node, publicport,
+                              privateport, protocol="tcp"):
+        params = {"cloudspaceId": int(network.id),
+                  "machineId": int(node.id),
+                  "localPort": privateport,
+                  "publicPort": publicport,
+                  "publicIp": network.publicipaddress,
+                  "protocol": protocol}
+        self._api_request("/portforwarding/create", params)
+        return self._to_port_forward(params, network)
+
+    def ex_delete_portforward(self, portforward):
+        params = {"cloudspaceId": int(portforward.network.id),
+                  "publicIp": portforward.network.publicipaddress,
+                  "publicPort": portforward.publicport,
+                  "proto": portforward.protocol}
+        self._api_request("/portforwarding/deleteByPort", params)
+        return True
+
+    def start_node(self, node):
+        """
+        Start virtual machine
+        """
+        node.state = NodeState.STARTING
+        self._api_request("/machines/start", {"machineId": int(node.id)})
+        node.state = NodeState.RUNNING
+        return True
+
+    def ex_list_networks(self):
+        """
+        Return the list of networks.
+
+        :return: A list of network objects.
+        :rtype: ``list`` of :class:`G8Network`
+        """
+        networks = []
+        for network in self._api_request("/cloudspaces/list"):
+            if network["accountId"] == self._account_id:
+                networks.append(self._to_network(network))
+        return networks
+
+    def list_sizes(self):
+        """
+        Returns a list of node sizes as a cloud provider might have
+
+        """
+        location = self._location["locationCode"]
+
+        sizes = []
+        for size in self._api_request("/sizes/list", {"location": location}):
+            sizes.extend(self._to_size(size))
+        return sizes
+
+    def list_nodes(self, network=None):
```

Comment:
``network`` argument is not part of the base API so please prefix it with ``ex_``.